pkg/apis/authorization/types.go

```diff
@@ -177,6 +177,37 @@ type SelfSubjectRulesReviewSpec struct {
 	Namespace string
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SubjectRulesReview is like SelfSubjectRulesReview, but can query any user, not just the current user.
+type SubjectRulesReview struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// Spec holds information about the request being evaluated.
+	Spec SubjectRulesReviewSpec
+
+	// Status is filled in by the server and indicates the set of actions a user can perform.
+	Status SubjectRulesReviewStatus
+}
+
+// SubjectRulesRevieSpec is a description of the rules review request.
+type SubjectRulesReviewSpec struct {
+	// Namespace to evaluate rules for. Required.
+	Namespace string
+
+	// User is the user you're testing for.
+	// If you specify "User" but not "Group", then is it interpreted as "What if User were not a member of any groups"
+	User string
```

Comment:
These four: user, group, extra, uid.  They should be the same as the ones in `SubjectAccessReviewSpec`.  @liggitt @smarterclayton is there any way to embed a `User struct` and keep the proto serialization for the existing `SubjectAccessReviewSpec` the same as it is today?